clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -345,7 +345,10 @@ public boolean equals(Object obj) {
             } else {
                 Object thisField = this.get(f);
                 Object otherField = other.get(f);
-                result = (thisField == null && otherField == null) || thisField.equals(otherField);
+                if (thisField == null)
+                    return otherField == null;
+                else
+                    return thisField.equals(otherField);
```

Comment:
The ternary operator is often used in such cases, any reason not to do that here?

```java
result = thisField == null ? otherField == null : thisFied.equals(otherField);